cli/tools/bench.rs

```diff
@@ -53,214 +49,311 @@ struct BenchSpecifierOptions {
-  pub iterations: u64,
-  pub current_start: Instant,
-  pub measures: Vec<u128>,
+#[derive(Debug, Clone, PartialEq, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub enum BenchResult {
+  Ok(BenchMeasurement),
+  Failed(BenchFailure),
 }
 
-#[derive(Debug, Clone)]
-pub struct BenchSummary {
+#[derive(Debug, Clone, Serialize)]
+pub struct BenchReport {
   pub total: usize,
-  pub passed: usize,
   pub failed: usize,
-  pub ignored: usize,
-  pub filtered_out: usize,
-  pub measured: usize,
-  pub measures: Vec<BenchMeasures>,
-  pub current_bench: BenchMeasures,
-  pub failures: Vec<(BenchDescription, String)>,
+  pub failures: Vec<BenchFailure>,
+  pub measurements: Vec<BenchMeasurement>,
 }
 
-impl BenchSummary {
+#[derive(Debug, Clone, PartialEq, Deserialize, Eq, Hash)]
+pub struct BenchMetadata {
+  pub name: String,
+  pub origin: String,
+  pub baseline: bool,
+  pub group: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct BenchMeasurement {
+  pub name: String,
+  pub baseline: bool,
+  pub stats: BenchStats,
+  pub group: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct BenchFailure {
+  pub name: String,
+  pub error: String,
+  pub baseline: bool,
+  pub group: Option<String>,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct BenchStats {
+  pub n: u64,
+  pub min: f64,
+  pub max: f64,
+  pub avg: f64,
+  pub p75: f64,
+  pub p99: f64,
+  pub p995: f64,
+  pub p999: f64,
+}
+
+impl BenchReport {
   pub fn new() -> Self {
     Self {
       total: 0,
-      passed: 0,
       failed: 0,
-      ignored: 0,
-      filtered_out: 0,
-      measured: 0,
-      measures: Vec::new(),
-      current_bench: BenchMeasures {
-        iterations: 0,
-        current_start: Instant::now(),
-        measures: vec![],
-      },
       failures: Vec::new(),
+      measurements: Vec::new(),
     }
   }
+}
 
-  fn has_failed(&self) -> bool {
-    self.failed > 0 || !self.failures.is_empty()
-  }
-
-  fn has_pending(&self) -> bool {
-    self.total - self.passed - self.failed - self.ignored > 0
-  }
+fn create_reporter(show_output: bool) -> Box<dyn BenchReporter + Send> {
+  Box::new(ConsoleReporter::new(show_output))
 }
 
 pub trait BenchReporter {
+  fn report_group_summary(&mut self);
   fn report_plan(&mut self, plan: &BenchPlan);
-  fn report_wait(&mut self, description: &BenchDescription);
+  fn report_end(&mut self, report: &BenchReport);
+  fn report_wait(&mut self, wait: &BenchMetadata);
   fn report_output(&mut self, output: &BenchOutput);
-  fn report_result(
-    &mut self,
-    description: &BenchDescription,
-    result: &BenchResult,
-    elapsed: u64,
-    current_bench: &BenchMeasures,
-  );
-  fn report_summary(&mut self, summary: &BenchSummary, elapsed: &Duration);
+  fn report_result(&mut self, result: &BenchResult);
 }
 
-struct PrettyBenchReporter {
-  echo_output: bool,
+#[allow(dead_code)]
+struct JSONReporter {}
```

Comment:
It looks like there's no CLI option to enable this reporter, please remove for now, or add a TODO comment